addons/mrp/models/mrp_workorder.py

```diff
@@ -142,25 +143,41 @@ def _read_group_workcenter_id(self, workcenters, domain, order):
         'finished_workorder_id', string='By-products')
     allowed_lots_domain = fields.One2many(comodel_name='stock.production.lot', compute="_compute_allowed_lots_domain")
 
-    # Both `date_planned_start` and `date_planned_finished` are related fields on `leave_id`. Let's say
-    # we slide a workorder on a gantt view, a single call to write is made with both
-    # fields Changes. As the ORM doesn't batch the write on related fields and instead
-    # makes multiple call, the constraint check_dates() is raised.
-    # That's why the compute and set methods are needed. to ensure the dates are updated
-    # in the same time.
-    @api.depends('leave_id')
-    def _compute_dates_planned(self):
+    @api.depends('state')
+    def _compute_calendar_date(self):
         for workorder in self:
-            workorder.date_planned_start = workorder.leave_id.date_from
-            workorder.date_planned_finished = workorder.leave_id.date_to
-
-    def _set_dates_planned(self):
-        date_from = self[0].date_planned_start
-        date_to = self[0].date_planned_finished
-        self.mapped('leave_id').write({
-            'date_from': date_from,
-            'date_to': date_to,
-        })
+            if workorder.state == 'done':
+                workorder.date_calendar_start = workorder.date_start
+                workorder.date_calendar_finished = workorder.date_finished
+            else:
+                workorder.date_calendar_start = workorder.date_planned_start
+                workorder.date_calendar_finished = workorder.date_planned_finished
+
+    def _set_calendar_date(self):
+        date_calendar_start = self[0].date_calendar_start
+        date_calendar_finished = self[0].date_calendar_finished
+        for workorder in self:
+            if workorder.state in ('pending', 'ready', 'cancel'):
+                workorder.write({
+                    'date_planned_start': date_calendar_start,
+                    'date_planned_finished': date_calendar_finished,
```

Comment:
i would not use an inverse field to write on other stored fields. to me, it makes more sense to move this code to an override of write

what do you think @amoyaux ?